language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/visitors/FunctionCallExpressionTypeFinder.java

```diff
@@ -414,8 +423,6 @@ public void visit(ReturnStatementNode returnStatementNode) {
         if (resultFound && returnTypeSymbol.typeKind() == TypeDescKind.FUNCTION) {
             FunctionTypeSymbol functionTypeSymbol = (FunctionTypeSymbol) returnTypeSymbol;
             functionTypeSymbol.returnTypeDescriptor().ifPresentOrElse(this::checkAndSetTypeResult, this::resetResult);
-        } else {
-            resetResult();
```

Comment:
Why is this deleted?

Reply:
There are currently two reasons to do this:

1. This resets the type symbol found in the worker node. While the function captures the function type symbol, this cannot be replicated to the worker (as there is no worker type symbol but rather a symbol). This broadens the possible values for the type symbol under the worker node, and hence, we cannot replicate the `returnTypeSymbol.typeKind() == TypeDescKind.FUNCTION` condition for the worker.

2. Looking at the existing logic and test cases, this line appears to be unreachable. AFAIK, the statement is there to reset the results for the negative paths. Currently, this class is only used by the `call function` CA, and it appears to narrow these negative paths, thus making this logic redundant.